### PR TITLE
fix(design-tokens): define unique tokens for each tokens provider instance - FE-4691

### PR DESCRIPTION
**Pull request by jhajduk-sage**
### Proposed behaviour
This PR introduces class generation for each theme separately, so they can be used with multiple CarbonProvider instances at the same time.

### Current behaviour

Currently css variable definitions are defined for one CarbonProvider instance.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

There should be possibility of using different carbon provider instances with different themes. please check CodeSandbox: https://codesandbox.io/s/carbon-quickstart-forked-yp50d

### CODE REVIEW

File: src/style/design-tokens/carbon-scoped-tokens-provider/carbon-scoped-tokens-provider.component.js [hunk 1]
```diff
@@ -10,24 +10,41 @@ import createGuid from "../../../__internal__/utils/helpers/guid";
  *
  */
 
-export const tokensClassName = `sageDesignTokens-${createGuid()}`;
+const activeThemes = {};
+
+const carbonInstanceId = createGuid();
+
+const snakeCase = (input) =>
```
Comment (DipperTheDan) [inline src/style/design-tokens/carbon-scoped-tokens-provider/carbon-scoped-tokens-provider.component.js:17]: I think this const might be confusing as this is snake_case and this is kebab-case. Calling this const `kebabCase` maybe better suited to the format that the function returns.

File: src/components/portal/portal.js [hunk 2]
```diff
@@ -46,7 +48,10 @@ const Portal = ({ children, className, id, onReposition }) => {
       setPortalNode(node);
     } else if (!node) {
       node = document.createElement("div");
-      node.classList.add(portalClassName, tokensClassName);
+      node.classList.add(
+        portalClassName,
+        tokensClassName((theme && theme.name) || "")
```
Comment (DipperTheDan) [inline src/components/portal/portal.js:53]: You could possibly change this to:
`tokensClassName((theme?.name) || "")`

**Review by DipperTheDan — COMMENTED** [review]
As well as the comments, I also think the commit message should be a `fix(design-tokens): ...` as this work fixes a bug and a `refactor():` is for a code change that doesn't fix a bug or doesn't add a feature.

File: src/style/design-tokens/carbon-scoped-tokens-provider/carbon-scoped-tokens-provider.component.js (hunk 1 above)
Comment (jhajduk-sage) [reply src/style/design-tokens/carbon-scoped-tokens-provider/carbon-scoped-tokens-provider.component.js:17]: Of course. That was just a brain fart ;)

**Review by DipperTheDan — DISMISSED** [review]

**Review by grabkowski — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Comment by harpalsingh** [comment]
I have passed UX review, but not really sure it was needed as this is a functional development change, but looks all good to me.

**Pull request merged**